### DISCUSSION TITLE
We are facing an issue related to Apex-Roll; when we unlink or remove the lookup field from a child record, the calculations on the parent record are not correct.

### DISCUSSION THREAD

**Discussion by Tushar-013** [jamessimone/apex-rollup#546] [Q&A]
### Description of Implementation and Issue:

We have a requirement to perform a roll-up calculation to determine the sum of the ‘Amount’ field for records in the ‘Temp Transaction’ object. This roll-up should be applied to corresponding fields on the ‘Temp Disclosure’ object, with variations based on the record types (A, B, and C) in the ‘Temp Transaction’ object. Specifically, the sum of amounts should be populated in different fields on the 'Temp Disclosure' object, depending on the record type.
Eg:
Sum of All Temp Transactions record of Record Type A ->  Sum of Record A Trnx field
Sum of All Temp Transactions record of Record Type B ->  Sum of Record B Trnx field
Sum of All Temp Transactions record of Record Type C ->  Sum of Record C Trnx field

**Roll Up Setup in Custom MetaData:**

We setup the Custom metadata for Rollup and create 3 manage records:
Rollup_Temp_Trnx_Amt_on_Sum_Record_A
Rollup_Temp_Trnx_Amt_on_Sum_Record_B
Rollup_Temp_Trnx_Amt_on_Sum_Record_C

<img width="1257" alt="Screenshot 2023-12-15 at 1 04 33 PM" src="https://github.com/jamessimone/apex-rollup/assets/53346280/187a980a-95a9-405a-a664-de3030f5c5b2">

**Trigger setup as per documentation:**

<img width="1402" alt="Screenshot 2023-12-15 at 1 55 26 PM" src="https://github.com/jamessimone/apex-rollup/assets/53346280/6ec4109a-6006-4857-b0d0-8b02de1ef173">

We create Temp Disclosure Record and attach it to the Temp transactions.

**Temp Disclosure Record:**

<img width="974" alt="Screenshot 2023-12-14 at 1 42 22 PM" src="https://github.com/jamessimone/apex-rollup/assets/53346280/a67af305-5a8b-416b-9c62-a6515bfadbbd">

**Temp Transaction Record:**

<img width="1401" alt="Screenshot 2023-12-14 at 1 43 01 PM" src="https://github.com/jamessimone/apex-rollup/assets/53346280/6bd7c348-d295-4580-82f0-9a732e639a81">

After removing the lookup field from the Child Object record (Temp Transaction Record) for Record Type B, the calculation is not proper. It subtracts the amount of Record Type B from all fields: Sum of Record A TRNX, Sum of Record B TRNX, and Sum of Record C TRNX. However, ideally, it should only subtract the amount from the 'Sum of Record B TRNX' field in Temp Disclosure because we have already specified the conditions in the custom metadata.

<img width="965" alt="Screenshot 2023-12-14 at 1 45 45 PM" src="https://github.com/jamessimone/apex-rollup/assets/53346280/d298b2ec-c8fc-4e83-b08d-2aef87bd0793">

Calculated Amount after the removing lookup field from Record Type B record of Temp Transaction.

**The calculated sum is incorrect for the 'Sum of Record A TRNX' and 'Sum of Record C TRNX'.**

<img width="505" alt="Screenshot 2023-12-15 at 2 26 54 PM" src="https://github.com/jamessimone/apex-rollup/assets/53346280/018ccb80-fc0a-4d31-9ffe-bc721d427911">

### **Can you please provide guidance regarding the issue or help us by suggesting some solutions on how we can fix this?**

**Comment by jamessimone** [comment]
I will take a look at this. Apologies for the delay in response!

**Reply by jamessimone** [reply]
@Tushar-013 I have not been able to reproduce this issue. I've created record types, validated that removing the parent from one record only modifies rollup fields with a matching where clause, tried all kinds of different where clauses, and haven't had any success. Have you tried enabling rollup logging using the Org Default rollup control record? Perhaps there's some kind of recursive process or other issue that is org-specific.
